app/controllers/api/base_controller.rb

```diff
@@ -302,7 +302,7 @@ def add_info_headers
     # e.g. /hosts/fqdn/reports # would add host = fqdn to the search bar
     def setup_search_options
       params[:search] ||= ""
-      params.keys.each do |param|
+      params.each_key do |param|
         if param =~ /(\w+)_id$/
           if params[param].present?
             query = " #{Regexp.last_match(1)} = #{params[param]}"
```

Comment:
I think it would better to avoid lookups if we already know the value:
```suggestion
      params.each do |param, value|
        if param =~ /(\w+)_id$/
          if value.present?
            query = " #{Regexp.last_match(1)} = #{value}"
```

Reply:
Submitted as https://github.com/theforeman/foreman/pull/10108 with some additional fixes for the `hash.keys.sort` equivalent.